metric.go

```diff
@@ -113,16 +113,51 @@ type MetricDataArray []*MetricData
 
 // for ES
 type MetricDefinition struct {
-	Id         string   `json:"id"`
-	OrgId      int      `json:"org_id"`
-	Name       string   `json:"name" elastic:"type:string,index:not_analyzed"` // graphite format
-	Metric     string   `json:"metric"`                                        // kairosdb format (like graphite, but not including some tags)
-	Interval   int      `json:"interval"`                                      // minimum 10
-	Unit       string   `json:"unit"`
-	Mtype      string   `json:"mtype"`
-	Tags       []string `json:"tags" elastic:"type:string,index:not_analyzed"`
-	LastUpdate int64    `json:"lastUpdate"` // unix timestamp
-	Partition  int32    `json:"partition"`
+	Id           string   `json:"id"`
+	OrgId        int      `json:"org_id"`
+	Name         string   `json:"name" elastic:"type:string,index:not_analyzed"` // graphite format
+	Metric       string   `json:"metric"`                                        // kairosdb format (like graphite, but not including some tags)
+	Interval     int      `json:"interval"`                                      // minimum 10
+	Unit         string   `json:"unit"`
+	Mtype        string   `json:"mtype"`
+	Tags         []string `json:"tags" elastic:"type:string,index:not_analyzed"`
+	LastUpdate   int64    `json:"lastUpdate"` // unix timestamp
+	Partition    int32    `json:"partition"`
+	NameWithTags string   `json:"nameWithTags"`
+}
+
+func NewMetricDefinition(name string, tags []string) *MetricDefinition {
+	sort.Strings(tags)
+
+	m := &MetricDefinition{Name: name, Tags: tags}
+	m.DeduplicateNameWithTags()
+
+	return m
+}
+
+// DeduplicateNameWithTags deduplicates the name and tags strings by storing
+// their content as a single string in .NameWithTags and then makes .Name and
+// the .Tags slices of it.
+func (m *MetricDefinition) DeduplicateNameWithTags() {
```

Comment:
probably doesn't need to be an exported function

Reply:
I think it should be actually.
There might be many cases where a system (f.e. MT) receives an MD and then wants to deduplicate that `Name` and `Tags` strings.

Reply:
right.